Hparg/BoxPlot.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Hparg.Drawable;
+using Hparg.Plot;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+
+namespace Hparg
+{
+    public class BoxPlot : APlot<float>
+    {
+        public BoxPlot(IEnumerable<float> data, Action<IEnumerable<float>> callback = null) : base(callback)
+        {
+            _data = data;
+            Min = _data.Min();
+            Max = _data.Max();
+        }
+
+        private readonly IEnumerable<float> _data;
+
+        internal override IEnumerable<float> GetPointsInRectangle(float x, float y, float w, float h)
+        {
+            return Array.Empty<float>();
+        }
+
+        private float GetQuarter(IOrderedEnumerable<float> _data, int parts, int index)
+        {
+            float count = _data.Count() / (float)parts * index;
+            if (count == (int)count) // Devide give round value so everything is okay
+            {
+                return _data.ElementAt((int)count);
+            }
+            return (_data.ElementAt((int)count) + _data.ElementAt((int)count)) / 2f;
```

Comment:
1) Should be : (_data.ElementAt((int)count) + _data.ElementAt((int)count + 1)) / 2f;
2) This formulae is true for median, but not for other quantiles: if you ask for the alpha% quantile you need to interpolate at alpha% between the two values.
i.e. : 1st Quartile is: (3 / 4f) * _data.ElementAt((int)count) + (1 / 4f) * _data.ElementAt((int)count + 1));

Reply:
Fixed 1. with b26604df888112c9393dee4fb300fd1c2f51ba47